src/operator/controllers/intents_reconcilers/iam/iampolicyagents/azurepolicyagent/agent.go

```diff
@@ -154,18 +144,29 @@ func (a *Agent) ensureRoleAssignmentsForIntents(ctx context.Context, userAssigne
 	return nil
 }
 
-func (a *Agent) ensureRoleAssignmentsForIntent(ctx context.Context, scope string, roleNames []string, userAssignedIdentity armmsi.Identity, existingRoleAssignmentsForScope []armauthorization.RoleAssignment) error {
+func (a *Agent) ensureRoleAssignmentsForIntent(
+	ctx context.Context,
+	scope string,
+	roleNames []string,
+	userAssignedIdentity armmsi.Identity,
+	existingRoleAssignmentsForScope []armauthorization.RoleAssignment,
+	intents otterizev2alpha1.ClientIntents,
+) error {
 	roleDefinitionsByName, err := a.FindRoleDefinitionByName(ctx, scope, roleNames)
 	if err != nil {
 		return errors.Wrap(err)
 	}
-
 	existingRoleDefinitionIDs := goset.FromSlice(lo.Map(existingRoleAssignmentsForScope, func(roleAssignment armauthorization.RoleAssignment, _ int) string {
 		return *roleAssignment.Properties.RoleDefinitionID
 	}))
 
 	for _, roleName := range roleNames {
-		roleDefinition := roleDefinitionsByName[roleName]
+		roleDefinition, exists := roleDefinitionsByName[roleName]
+		if !exists {
+			// The role mentioned in the intents does not exist in the subscription, record event and move on
+			a.RecordWarningEventf(&intents, ReasonRoleDefinitionNotFound, "Role definition %s not found in scope %s", roleName, scope)
```

Comment:
```suggestion
			a.RecordWarningEventf(&intents, ReasonRoleNotFound, "Role %s not found in scope %s", roleName, scope)
```
From the user's perspective he's specifying roles, not role definitions (in the ClientIntents), so the error should use the same terminology